File: clients/python/model/building_type.py
from enum import IntEnum

class BuildingType(IntEnum):
    """TODO - Document"""

    QUARRY = 0
    """TODO - Document"""
    MINES = 1
    """TODO - Document"""
    CAREER = 2
    """TODO - Document"""
    FARM = 3
    """TODO - Document"""
    FOUNDRY = 4
    """TODO - Document"""
    FURNACE = 5
    """TODO - Document"""
    BIOREACTOR = 6
    """TODO - Document"""
    CHIP_FACTORY = 7
    """TODO - Document"""
    ACCUMULATOR_FACTORY = 8
    """TODO - Document"""
    REPLICATOR = 9
    """TODO - Document"""

    def __repr__(self):
        return str(self)

File: clients/rust/src/model/building_type.rs
use super::*;

/// TODO - Document
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub enum BuildingType {
    /// TODO - Document
    Quarry,
    /// TODO - Document
    Mines,
    /// TODO - Document
    Career,
    /// TODO - Document
    Farm,
    /// TODO - Document
    Foundry,
    /// TODO - Document
    Furnace,
    /// TODO - Document
    Bioreactor,
    /// TODO - Document
    ChipFactory,
    /// TODO - Document
    AccumulatorFactory,
    /// TODO - Document
    Replicator,
}

impl trans::Trans for BuildingType {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        let tag: i32 = match self {
            Self::Quarry => 0,
            Self::Mines => 1,
            Self::Career => 2,
            Self::Farm => 3,
            Self::Foundry => 4,
            Self::Furnace => 5,
            Self::Bioreactor => 6,
            Self::ChipFactory => 7,
            Self::AccumulatorFactory => 8,
            Self::Replicator => 9,
        };
        trans::Trans::write_to(&tag, writer)
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let tag = <i32 as trans::Trans>::read_from(reader)?;
        match tag {
            0 => Ok(Self::Quarry),
            1 => Ok(Self::Mines),
            2 => Ok(Self::Career),
            3 => Ok(Self::Farm),
            4 => Ok(Self::Foundry),
            5 => Ok(Self::Furnace),
            6 => Ok(Self::Bioreactor),
            7 => Ok(Self::ChipFactory),
            8 => Ok(Self::AccumulatorFactory),
            9 => Ok(Self::Replicator),
            _ => Err(std::io::Error::new(
                std::io::ErrorKind::Other,
                format!("Unexpected tag {:?}", tag))),
        }
    }
}